File: Week1/henry/1-3.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
bool comp(string &a, string &b)
{
	//printf("*");
	int i = 0, j = 0, t1 = 0, t2 = 0;
	char t;
	string s1 = "", s2 = "";
	while (!('0' <= a[i] && a[i] <= '9')) //헤드 파트
	{
		t = a[i];
		if ('A' <= t && t <= 'Z')
			t += 32;
		s1 += t;
		i++;
	}
	while (!('0' <= b[j] && b[j] <= '9'))
	{
		t = b[j];
		if ('A' <= t && t <= 'Z')
			t += 32;
		s2 += t;
		j++;
	}
	if (s1 != s2)
		return s1 < s2;
	while (('0' <= a[i] && a[i] <= '9')) //숫자 파트
	{
		t1 *= 10;
		t1 += a[i] - '0';
		i++;
	}
	while (('0' <= b[j] && b[j] <= '9'))
	{
		t2 *= 10;
		t2 += b[j] - '0';
		j++;
	}
	return t1 <= t2;
}
vector<string> solution(vector<string> files)
{
	string t;
	for (int i = 0; i < files.size(); ++i) //버블 정렬
		for (int j = 0; j < files.size() - 1 - i; ++j)
			if (!comp(files[j], files[j + 1]))
			{
				t = files[j];
				files[j] = files[j + 1];
				files[j + 1] = t;
			}
	//sort(files.begin(), files.end(), comp);
	return files;
}

File: Week1/jihee/1-3.py
# 파일명 정렬
def solution(files):
    answer = []
    new = []
    
    for i in range(len(files)):
        k=0
        tmp = files[i][k]
        # 숫자 전까지 head에 저장
        while not tmp.isdigit():
            k += 1
            tmp = files[i][k]
        head = files[i][0:k].lower()
        index_digit = k
        # number 부분 저장
        while tmp.isdigit():
            k += 1
            if len(files[i])<=k:
                break
            tmp = files[i][k]
        number = int(files[i][index_digit:k])
        # 파일의 입력 순서 저장
        tail = i
        # 새로운 list new에 head, number, 입력 순서가 담긴 tuple 저장
        new.append((head, number, tail))

    # head, number, 입력순서 순의 우선순위로 정렬
    sorting = sorted(new, key = lambda x: (x[0], x[1], x[2]))
    
    # 정렬된 순서대로 answer에 원래의 file 명 apppend
    for i in range(len(files)):
        answer.append(files[sorting[i][2]])

    return answer

if __name__ == "__main__":
    files = ["MUZI02", "muzi1"]
    answer = solution(files)
    print(answer)